zlib.h

```diff
@@ -1645,7 +1645,7 @@ ZEXTERN void ZEXPORT gzclearerr(gzFile file);
    library.
 */
 
-ZEXTERN uint32_t ZEXPORT adler32(uint32_t adler, const unsigned char *buf, uint32_t len);
+ZEXTERN unsigned long ZEXPORT adler32(unsigned long adler, const unsigned char *buf, unsigned int len);
```

Comment:
This seems a bit mysterious.  Won't it be wrong now on 64 bits in the non-compat case?

Reply:
Seems like we need a type that is ZNG_uLong which expands to 32 bits for ng, and unsized for compat?

Reply:
Non-compat case will not use this header, it uses zlib-ng.h instead.

Reply:
Using ZNG_uLong isn't enough as some compilers will issue warning or error if function parameters or return type is truncated. Explicit zero-extending is for readability mostly as it makes it clear for human reader and obvious for compiler when inlining -- and static analysis -- that the zero-extending is intended behavior and not oversight.